cmd/frontend/auth/non_public.go

```diff
@@ -111,6 +124,10 @@ func AllowAnonymousRequest(req *http.Request) bool {
 		return true
 	}
 
+	if checkAllowAnonymousRequest(req) {
+		return true
+	}
+
```

Comment:
Should we combine this with the other clauses? Also it seems like `checkAllowAnonymousRequest` is already doing the `conf.AuthPublic` check... E.g. something like:
```suggestion
	return checkAllowAnonymousRequest(req) || strings.HasPrefix(req.URL.Path, "/.assets/") || defaultCode
```

Reply:
Personally, I like to keep clauses separate for simplicity purposes, as combining them will make reading it harder. Besides all the rest of the existing clauses follow the same convention of being separate.